### DISCUSSION TITLE
Multiple Subscriptions

### DISCUSSION THREAD

**Discussion by simon-metcalf-cubic** [mivano/azure-cost-cli#41] [Ideas]
I needed to pull data for over 100 subscriptions so did a quick PowerShell script to loop through, this may be useful for others.

#Connect to Azure
Connect-AzAccount -Tenant xxxxx-xxxxx-xxxxx-xxxxx

#Pull list of subscriptions
$ids = Get-AzSubscription -TenantId xxxxx-xxxxx-xxxxx-xxxxx | Select-Object id,name

#Loop through each subscription and create single CSV
Foreach ($id in $ids) {
$output = azure-cost costByResource -s $($id.id) -t Custom --from 2023-05-01 --to 2023-05-31 -o csv
Add-Content ".\report.csv" $output
}

**Comment by mivano** [comment]
Thanks for sharing! I think this adds the header in the csv multiple times. Would an optional parameter skipHeader be useful here? Setting it to true when it is not the first iteration?

**Reply by simon-metcalf-cubic** [reply]
Yes it did do that, yes optional would be good

**Reply by mivano** [reply]
Check, it is added in version 0.20.0!

**Reply by simon-metcalf-cubic** [reply]
Great work thanks for quick turnaround!!

I upgraded and tested and found this skipped headers for all so updated PowerShell script to write header only on first object.

#Connect to Azure
Connect-AzAccount -Tenant xxxxx-xxxxx-xxxxx-xxxxx

#Pull list of Subscriptions
$ids = Get-AzSubscription -TenantId xxxxx-xxxxx-xxxxx-xxxxx | Select-Object id,name
$firstid = $ids | Select-Object -first 1
$remainingids = $ids | Select-Object -skip 1

#Create CSV for first subscription with header
$firstoutput = azure-cost costByResource -s $firstid.id -t Custom --from 2023-05-01 --to 2023-05-31 -o csv
Add-Content ".\.\report.csv.csv" $firstoutput

#Loop through remaining subscriptions and append to CSV
Foreach ($id in $remainingids) {
$output = azure-cost costByResource -s $($id.id) -t Custom --from 2023-05-01 --to 2023-05-31 -o csv --skipHeader
Add-Content ".\.\report.csv.csv" $output
}

**Comment by mivano** [comment]
I added the solution to iterate over multiple subscriptions to the readme file.
